MID Server/MID Server Script Includes/IllumioHTTPClient.js

```diff
@@ -152,19 +161,15 @@ IllumioHTTPClient.prototype = {
      * @returns {JSON} Object of the response
      **/
     get: function(endpoint, queryString, headers, requestData) {
-        var retryCount = 0;
         var response = null;
-        while (retryCount < MAX_RETRY_COUNTS) {
-            var getMethod = new GetMethod(this.baseUrl + endpoint);
-            this.logger._debug("APICALL: GET endpoint: " + endpoint + " queryString: " + queryString);
-            response = this._executeMethod(endpoint, queryString, getMethod, headers, requestData);
-            if (!response.hasError) {
-                this.logger._debug("HTTP GET Call complete, status: " + response.status);
-                return response;
-            }
-            this.logger._error("Failed: HTTP GET Call, status: " + response.status);
-            retryCount++;
+        var getMethod = new GetMethod(this.baseUrl + endpoint);
+        this.logger._debug('APICALL: GET endpoint: ' + endpoint + ' queryString: ' + queryString);
+        response = this._executeMethod(endpoint, queryString, getMethod, headers, requestData, this.HTTP_RETRY_COUNT);
+        if (!response.hasError) {
+            this.logger._debug('HTTP GET Call complete, status: ' + response.status);
```

Comment:
There isn't a need to duplicate these `hasError` checks and debug lines in each CRUD function; we can call it once in `_executeMethod`.

Reply:
All 4 has different methods, so just to give the log i have added this line, else i will have to pass another argument into the function

Reply:
You're already passing the method (the GetMethod/PostMethod/etc class). You can use `method.getName()` to get the method name in `_executeMethod` for the logs.